File: yh_star/scripts/yh_star_sub.py
#!/usr/bin/python3
# -*- coding: utf-8 -*- 

import rospy
from yh_star.msg import YhStarMsg

def msgCallback(msg):
    n = msg.data

    for i in range(1, n//2+1):
        for j in range(i):
            print("*", end= '')
        print()
    for i in range(n//2+1, n+1):
        for j in range(n-i+1):
            print("*", end= '')
        print()

def star_printer():
    rospy.init_node("yh_star_sub")
    rospy.Subscriber("yh_star_topic",YhStarMsg,msgCallback)
    rospy.spin()

if __name__ == "__main__":
    star_printer()

File: yh_star/src/yh_star_sub.cpp
#include "ros/ros.h"
#include "yh_star/YhStarMsg.h"  


//void msgcallback(const std_msgs::String::ConstPtr& msg)
void msgCallback(const yh_star::YhStarMsg::ConstPtr& msg)  //어디서 꺼내오는지가 바뀜.
{
    int n = msg->data;
    for (int i=0; i<n; i++)
    {
       for(int j = 0; j < i+1; j++)
       {
            printf("*");
       }
       printf("\n");
    }
    for (int i =n/2; i < n; i++)
    {
        for(int j = n; j > i; j--)
        {
            printf("*");
        }
        printf("\n");
    }
    
}

int main(int argc, char** argv)
{
    ros::init(argc,argv, "yh_star_sub");
    ros::NodeHandle nh;

    ros::Subscriber sub = nh.subscribe("yh_star_topic",30,msgCallback);

    ros::spin(); // 대기
    
    return 0;
}